server/http/src/service.rs

```diff
@@ -166,17 +166,45 @@ where
         self.serve_dynamic_file(self.graphiql_html())
     }
 
+    fn resolve_version_number(
+        &self,
+        request: &Request<Body>,
+    ) -> Result<VersionNumber, GraphQLServerError> {
+        let version_header = request.headers().get("api-version");
+
+        let mut version = VersionNumber::default();
+
+        if let Some(header_value) = version_header {
+            let version_value = header_value.to_str().map_err(|_| {
+                GraphQLServerError::ClientError(format!(
+                    "Invalid api-version header value {:?}",
+                    header_value
+                ))
+            })?;
+            VersionNumber::validate(version_value.to_string())
+                .map_err(|error| GraphQLServerError::ClientError(error))?;
+
+            version = version_value.into();
+        }
+
+        Ok(version)
+    }
```

Comment:
I used `api-version` header because it was the easiest to implement quickly but before merging the Pull Request we should switch to url-based approach (this way we could use GraphiQL - it does not support headers).